Update ActiveRecord callback guide for exceptions

naw opened the pull request:
07d3d402341e81ada0214f2cb2be1da69eadfe72 changed how exceptions are handled in `after_commit` and `after_destroy` callbacks.  This PR corrects the guide to reflect the new behavior.

I'm assuming this should go in `master` _and_ `5-0-stable`. Not sure if I should open separate PR's?

vipulnsward commented on guides/source/5_0_release_notes.md:
```diff
@@ -453,6 +453,13 @@ Please refer to the [Changelog][action-pack] for detailed changes.
+
+    Now these errors are not rescued anymore and just bubble up, as the other callbacks.
+    ([commit](https://github.com/rails/rails/commit/07d3d402341e81ada0214f2cb2be1da69eadfe72))
 
```
Can you revert changes in this file, it should not be updated.

gladtocode committed:
Update docs for change in transaction callback exceptions

07d3d40 changed how exceptions are handled in after_commit
and after_destroy callbacks.  This commit updates the 5.0
release notes and the  ActiveRecord callback guide to reflect
the new behavior.

[ci skip]

vipulnsward commented:
@naw change looks good, just one note, nothing should be changed in the release notes file.

naw commented:
@vipulnsward I would be happy to revert, but how else will people who are upgrading to Rails 5 be aware of this breaking change, unless they read through the entire ActiveRecord changelog?

vipulnsward commented:
Actually my mistake, I thought this was a change after 5.0 release.
In that case 👍 for the change.

vipulnsward commented on guides/source/5_0_release_notes.md:
```diff
@@ -797,6 +797,14 @@ Please refer to the [Changelog][active-record] for detailed changes.
+*   Change transaction callbacks to not swallow errors.
+    Before this change any errors raised inside a transaction callback
+    were getting rescued and printed in the logs, unless you used
+    the (newly deprecated) `raise_in_transactional_callbacks = true` option
```
`.` after option.

naw replied:
I have added the `.`

naw commented:
Right, sorry, I didn't make it very clear in my PR message when this behavior actually changed.

vipulnsward commented:
Thanks!
